### DISCUSSION TITLE
wasm backend ?

### DISCUSSION THREAD

**Discussion by rraammiinn** [imba/imba#831] [Q&A]
if someday wasm be able to manipulate dom directly could imba compile to wasm using some sort of gradual type system ?

**Comment by haikyuu** [comment]
There is some work to run js in WASM https://github.com/Shopify/javy so it's quite straightforward.
But I bet js in the browser is still faster than WASM (not sure though).

I see that useful outside the browser for example to integrate with systems that don't speak JS like a database or creating a plugin ...

**Reply by rraammiinn** [reply]
not that way .
typescript and imba both support static typing but only for its strictness and they don't take advantage of it for performance purposes .

**Comment by rraammiinn** [comment]
do imba compiles also to typescript ?
